girder_worker/docker/transforms/girder.py

```diff
@@ -298,3 +298,9 @@ def transform(self, *args, **kwargs):
     def exception(self):
         if self._upload_on_exception:
             return self.transform()
+
+
+class ChainedResultItem(ChainedResultTransform):
+    def transform(self, results):
+        item_id = super(ChainedResultItem, self).transform(results)
```

Comment:
oh i see... do you have multiple results being returned from the first docker task? What are they?

Reply:
I'm not sure how this actually works, but my understanding of girder_result_hooks was that the return value from a docker_run task was always going to be a list, and I thought it was a list whose values are the resolved result hooks. So in the right-hand task in a chain, we need to be able to select which index into that list is the result we want to use.

I have no idea whether this will actually work. If you have an architecture you think is better, that would be great, but you'd need to write it out in code for me to understand it probably. The code inside `test_docker_run_multistep_workflow` in this PR is where we can show how this looks from an API perspective.